docs/en/03_usage.md

```diff
@@ -58,9 +58,80 @@ to happen separately from the indexing of content. This can include operations l
 creating indexes, and defining a schema for an index.
  
 Indexing services are required to define a `configure()` method, and this method is invoked
-during the `dev/build` process, as well as in the `SearchConfigure` task, for a more direct
+during the `db:build` process, as well as in the `tasks:SearchConfigure` task, for a more direct
 approach.
- 
+
+## Extension points
+There are multiple extension points in this module but some are only relevant to provider implementaion or deeper [customisations](./05_customising.md). The following are the more commonly used extension points provided by the module's DataObject implementation
+
+### DataObject extension points
+
+#### canIndexInSearch(): bool
+
+A DataObject or extension implementing this method can add supplementary logic to the [shouldIndex()](../../src/DataObject/DataObjectDocument.php#L128) method for determining whether a record should be indexed. More control can be had via the [IndexableHandler interface](./08_customising_more.md#indexablehandler-interface)
+
+### updateSearchDependentDocuments(&$dependentDocs): void
+
+A DataObject extension implementing this method can add dependent documents to its dependency list. This is particularly relevant if you're not using `auto_dependency_tracking`. It is important to remember that `$dependentDocs` in this context should be first-class `DocumentInterface` instances, not DataObjects.
```

Comment:
It might be useful to include an example here. It is not totally clear to me what I'd need to do to set this up. I'm guessing you mean that if you have a DataObject dependency that needs updating you'd do something like `$dataDocument = DataObjectDocument::create($dataObject);` and pass that.